src/stactools/naip/utils.py

```diff
@@ -5,7 +5,9 @@
 
 import dateutil.parser
 
-NAIP_FILENAME_REGEX = re.compile(r"(m)_(\d{7})_(\w{2})_(\d{2})_(\d{1,3})_(\d{8})")
+NAIP_FILENAME_REGEX = re.compile(
+    r"(m)_(\d{7})_(\w{2})_(\d{2})_(\w{1,3})_(\d{8})(_\d{8})?"
```

Comment:
the `m` doesn't need to be captured, so you can remove the parens around it. Any of the other groups that aren't used can be made non-capturing with `?:` (e.g., `(?:\d{7})`), which also requires changing the index of the groups being used for the id. One way to make that better would be to make that a named captured group like `(?P<some_name>\d{8})` and use it like `group("some_name")` (see https://docs.python.org/3/howto/regex.html#non-capturing-and-named-groups)

Reply:
updated